src/mapgen_v7.cpp

```diff
@@ -483,25 +480,34 @@ void MapgenV7::carveRivers() {
 
 int MapgenV7::generateTerrain()
 {
-	int ymax = generateBaseTerrain();
+	int stone_surface_max_y = -MAP_GENERATION_LIMIT;
+	bool generate_mountain = false;
+
+	generateBaseTerrain(&stone_surface_max_y, &generate_mountain);
+
+	if ((spflags & MGV7_MOUNTAINS) && generate_mountain) {
```

Comment:
Perhaps, but i can't think of anything nice and short, i think 'generateTerrain' is okay and not innaccurate.